### DISCUSSION TITLE
Pull through cache with non-docker registry backend (Artifactory)

### DISCUSSION THREAD

**Discussion by amitkumar2283** [distribution/distribution#4386] [General]
We have an Artifactory registry, but we cannot expose it to external customers.
Hence we want to use Distribution as a frontend cache and serve images from Artifactory via Pull Through cache.

But it looks like Distribution is NOT pulling images from Artifactory. It does work with `registry-1.docker.com` though.

**Comment by JStevenKalangi** [comment]
I too have same issue: When I looked at container logs, I notice that the registry tries to find the image under `Storage` filesystem/s3 and returns 404 but doesnt proceed with proxy url. I have certs included as part of image and pulling an image from S3 works fine.

trace.duration=307µs trace.file="github.com/docker/distribution/registry/storage/driver/base/base.go" trace.func="github.com/docker/distribution/registry/storage/driver/base.(*Base).GetContent" trace.id=f1b53d2f-4b54-44dd-ad6a-fcbcd234cd21 trace.line=95 vars.name="mysql" vars.reference=0.0.2 


My config file looks like this:
```
configData:
  version: 0.1
  log:
    level: debug
    fields:
      service: registry
  storage:
    cache:
      blobdescriptor: inmemory
  proxy:
    remoteurl: "https://<proxy_url>"
    username: "<myusername>"
    password: "<mypassword>"
  http:
    addr: :5000
    headers:
      X-Content-Type-Options: [nosniff]
    debug:
      addr: :5001
      prometheus:
        enabled: false
        path: /metrics
  health:
    storagedriver:
      enabled: true
      interval: 10s
      threshold: 3
```
I ve read the below 
"""
Docker offers [Registry as a pull through cache](https://docs.docker.com/registry/recipes/mirror/), unfortunately it only covers the DockerHub case. It won't cache images from quay.io, k8s.gcr.io, gcr.io, or any such, including any private registries.
""" at https://github.com/rpardini/docker-registry-proxy/blob/master/README.md.

Can you please suggest if distribution works with proxy URLs pointing to nexus

**Comment by amitkumar2283** [comment]
Is there anyone who can provide some insights?
